themes/powerline-naked/powerline-naked.base.bash

```diff
@@ -1,33 +1,36 @@
+# shellcheck shell=bash
 . "$BASH_IT/themes/powerline/powerline.base.bash"
 
 function __powerline_left_segment {
-  local OLD_IFS="${IFS}"; IFS="|"
-  local params=( $1 )
-  IFS="${OLD_IFS}"
-  local separator=""
-  local pad_before_segment=" "
+	local OLD_IFS="${IFS}"
+	IFS="|"
+	local params=("$1")
+	IFS="${OLD_IFS}"
+	local separator=""
+	local pad_before_segment=" "
 
-  if [[ "${SEGMENTS_AT_LEFT}" -eq 0 ]]; then
-    if [[ "${POWERLINE_COMPACT_BEFORE_FIRST_SEGMENT}" -ne 0 ]]; then
-      pad_before_segment=""
-    fi
-  else
-    if [[ "${POWERLINE_COMPACT_AFTER_SEPARATOR}" -ne 0 ]]; then
-      pad_before_segment=""
-    fi
-    # Since the previous segment wasn't the last segment, add padding, if needed
-    #
-    if [[ "${POWERLINE_COMPACT_BEFORE_SEPARATOR}" -eq 0 ]]; then
-      LEFT_PROMPT+=" "
-    fi
-    LEFT_PROMPT+="${POWERLINE_LEFT_SEPARATOR}"
-  fi
+	if [[ "${SEGMENTS_AT_LEFT}" -eq 0 ]]; then
+		if [[ "${POWERLINE_COMPACT_BEFORE_FIRST_SEGMENT}" -ne 0 ]]; then
+			pad_before_segment=""
+		fi
+	else
+		if [[ "${POWERLINE_COMPACT_AFTER_SEPARATOR}" -ne 0 ]]; then
+			pad_before_segment=""
+		fi
+		# Since the previous segment wasn't the last segment, add padding, if needed
+		#
+		if [[ "${POWERLINE_COMPACT_BEFORE_SEPARATOR}" -eq 0 ]]; then
+			LEFT_PROMPT+=" "
+		fi
+		LEFT_PROMPT+="${POWERLINE_LEFT_SEPARATOR}"
+	fi
 
-  LEFT_PROMPT+="$(set_color ${params[1]} -)${pad_before_segment}${params[0]}${normal}"
-  LAST_SEGMENT_COLOR=${params[1]}
-  (( SEGMENTS_AT_LEFT += 1 ))
+	LEFT_PROMPT+="$(set_color "${params[1]}" -)${pad_before_segment}${params[0]}${normal}"
+	LAST_SEGMENT_COLOR=${params[1]}
+	((SEGMENTS_AT_LEFT += 1))
+	_save-and-reload-history "${HISTORY_AUTOSAVE:-0}"
```

Comment:
This isn't needed here, since naked theme executes `__powerline_prompt_command()` from `powerline.base.bash` which already has the function added.